stellar-contract-env-host/src/test.rs

```diff
@@ -124,6 +124,82 @@ fn map_put_has_and_get() -> Result<(), HostError> {
     Ok(())
 }
 
+/// map test
+#[test]
+fn map_prev_and_next() -> Result<(), HostError> {
+    let host = Host::default();
+    let scmap: ScMap = vec![
+        ScMapEntry {
+            key: ScVal::U32(1),
+            val: ScVal::U32(2),
```

Comment:
Will this work for more complex key types, such as a key which is a Vec or a Map?

Reply:
Yes, these work under the `cmp` rules for `RawVal`. 
I've added a new more complex example in the test.